uk.co.compucorp.civicrm.hrleaveandabsences/js/angular/src/leave-absences/shared/components/leave-request-popup-files-tab.js

```diff
@@ -84,11 +89,121 @@ define([
       return !attachment.attachment_id || vm.canManage;
     };
 
+    vm.$onDestroy = unsubscribeFromEvents;
+
     (function init () {
-      vm.supportedFileTypes = _.keys(sharedSettings.fileUploader.allowedMimeTypes);
-      vm.request.loadAttachments();
+      $rootScope.$broadcast('LeaveRequestModal::childProcess::register');
+      initListeners();
+
+      $q.all([
+        loadSupportedFileTypes(),
+        vm.request.loadAttachments()
+      ])
+      .then(function () {
+        $rootScope.$broadcast('LeaveRequestModal::requestObjectUpdated');
+      })
+      .finally(function () {
+        vm.filesLoaded = true;
+      });
     }());
 
+    /**
+     * Returns an array of files marked for deletion
+     *
+     * @return {Array}
+     */
+    function filesMarkedForDeletion () {
+      return _.filter(vm.request.files, function (file) {
+        return file.toBeDeleted;
+      });
+    }
+
+    /**
+     * Returns the file uploader queue
+     *
+     * @return {Array}
+     */
+    function fileUploaderQueue () {
+      return (vm.fileUploader && vm.fileUploader.queue)
+        ? vm.fileUploader.queue : [];
+    }
+
+    /**
+     * Initializes all the listeners
+     */
+    function initListeners () {
+      listeners.push($rootScope.$on('LeaveRequestModal::childProcess::start', uploadAttachments));
```

Comment:
I think the current name is also meaningful. Because we want the childComponent to start processing so its okay to say `LeaveRequestModal::childComponent::start`, and the child processing logically can start at any time but only after submit. So keeping the current name is future proof. Also the components reply with `LeaveRequestModal::childComponent::success/error` which is also inline with the current naming convention.